components/feature/push/src/main/java/mozilla/components/feature/push/AutoPushFeature.kt

```diff
@@ -340,53 +323,18 @@ class AutoPushFeature(
     }
 }
 
-/**
- * Catches all known non-fatal push errors logs.
- */
-internal fun CoroutineScope.launchAndTry(
-    block: suspend CoroutineScope.() -> Unit,
-    errorBlock: (Exception) -> Unit
-): Job {
-    return launch {
-        try {
-            block()
-        } catch (e: RustPushError) {
-            val result = when (e) {
-                is GeneralError,
-                is CryptoError,
-                is CommunicationError,
-                is CommunicationServerError,
-                is AlreadyRegisteredError,
-                is StorageError,
-                is MissingRegistrationTokenError,
-                is StorageSqlError,
-                is TranscodingError,
-                is RecordNotFoundError,
-                is UrlParseError -> false
-                else -> true
-            }
-
-            if (result) {
-                throw e
-            }
-
-            errorBlock(e)
-        }
+internal inline fun exceptionHandler(crossinline onError: (PushError) -> Unit) = CoroutineExceptionHandler { _, e ->
+    val isFatal = when (e) {
+        is PushError.MalformedMessage,
+        is GeneralError,
+        is CryptoError,
+        is CommunicationError,
+        is CommunicationServerError -> false
+        else -> true
     }
-}
 
-/**
- * This is manager mapping of service type to channel ID, it will eventually be replaced by the
- * Application Service implementation.
- */
-internal object DeliveryManager {
-    /**
-     * Executes the block if the Push Manager is initialized.
-     */
-    fun with(connection: PushConnection, block: PushConnection.() -> Unit) {
-        if (connection.isInitialized()) {
-            block(connection)
-        }
+    if (isFatal) {
+        onError(PushError.Rust(e, e.message.orEmpty()))
```

Comment:
maybe at least log the non-fatal ones?

Reply:
This is the extension function, at the calling site, we use the logger there.